pkg/backend/openshiftcluster/adminupdate.go

```diff
@@ -5,13 +5,32 @@ package openshiftcluster
 
 import (
 	"context"
+	"time"
 
 	"github.com/Azure/ARO-RP/pkg/cluster"
+	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 func (m *Manager) AdminUpdate(ctx context.Context) error {
-	// m.ocDynamicValidator.Dynamic is not called so that it doesn't block an
-	// admin update
+	var err error
+	timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+
+	wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+		err = m.ocDynamicValidator.Dynamic(ctx)
+		if azureerrors.HasAuthorizationFailedError(err) ||
+			azureerrors.HasLinkedAuthorizationFailedError(err) {
+			m.log.Print(err)
+			return false, nil
+		}
+		return err == nil, err
+	}, timeoutCtx.Done())
+
+	if err != nil {
+		// don't block update on dynamic validator error.
```

Comment:
we don't want to block ourselves from being able to run update. However, I think there could be cases when we'd better be aware if there is broken configuration before we proceed with update. I suggest to add a "force" flag to have the best of the two worlds. First we'd run "soft" update which will fail if dynamic validator fails and if we are confident that update is safe regardless, we can do second update with force flag set.

Reply:
Do you want to do this as part of this work or separate?

Reply:
I prefer to do it separately to make the changes smaller. Right now this implementation doesn't change the behavior - update should proceed even if we encounter error during dynamic validator. However the problem is that we will see success in GA, and the error if exists, will be only visible in the logs.

Reply:
If we don't invoke it somewhere the whole PR is pointless. The additional checks that I have added are not possible at the time of creation, at least not when you create with az cli.

Reply:
I thought the idea here was that a separate Geneva Action was going to be written to run this code?